### ISSUE TITLE
 AWS - Auth social login, facebook and g+ session retain in browser

### ISSUE THREAD

**Issue by faizanmirza** [aws-amplify/aws-sdk-android#524]
1) When user is logged in app using customtab, facebook and google session is set in chrome browser. Next time when i logout and sign up again. It logged in with same user again instead of asking for new user. As per requirement for every new sign up on social sign up. It should ask for which account should be used to login instead of saved user session. 

2) Is it possible to use app webview instead of custom tab for aws-android-sdk-cognitoauth?

**Comment by kvasukib** [comment]
@faizanmirza The current implementation of `aws-android-sdk-cognitoauth` uses Chrome CustomTabs. Are you using isAuthenticated() method to check if the user is signed in before prompting the Web UI? If so, can you upgrade to the latest version of the SDK. We fixed a bug in the latest version of SDK where isAuthenticated returned true for a signed out user.

**Comment by jia-tse-zocdoc** [comment]
@kvasukib -- what if the devices doesn't have Chrome installed, or has it disabled in favor of their favorite browser? Is it possible to use app webview instead of custom tabs?

**Comment by harshalijain** [comment]
I am also facing the same issue. Even after calling cognitoAuth.signOut(), when i try to login through google again, I am not able to select another user.

**Comment by kashif-mujeeb-10p** [comment]
as per the requirement of @faizanmirza when the app is signup again it doesn't remove a cache that gonna be the wrong user experience, how could we achieve that?

**Comment by minbi** [comment]
Chrome Custom Tabs is more secure to use but offers less flexibility. I will discuss with the team on how we can clear the Google/Facebook sessions from the browser when signed-out.

**Comment by AlexeyYal** [comment]
I am facing this issue too. Could you use incognito mode for chrome here or use some another solution?

**Comment by ezimnitskiy** [comment]
Hi @minbi @frankmuellr, I have the same situation on my project. What to do? Please help, the deadline is close, but the problem is not solved.

**Comment by julianfalcionelli** [comment]
Yes, I am having this issue too, even after I logged out from my Facebook account in Chrome and manually cleared the cache, and login in with a different Facebook account the AWS SDK is still using the first account... 

```
            awsClient   : 'com.amazonaws:aws-android-sdk-mobile-client:2.16.12',
            awsUserPools: 'com.amazonaws:aws-android-sdk-auth-userpools:2.16.12',
            awsUserUI   : 'com.amazonaws:aws-android-sdk-auth-ui:2.16.12',
            awsCognito  : 'com.amazonaws:aws-android-sdk-cognitoauth:2.16.12'
```

**Comment by AlexZvonik** [comment]
I have the same problem. Does anybody know if any workaround available?

**Comment by phanisaripalli** [comment]
Is this issue resolved or is there a workaround? This is contributing to poor and confusing experience. Thanks.

**Comment by raphkim** [comment]
This issue is now resolved with [v2.24.0](https://github.com/aws-amplify/aws-sdk-android/releases/tag/release_v2.24.0). Please follow the [updated documentation](https://docs.amplify.aws/sdk/auth/hosted-ui/q/platform/android#setup-amazon-cognito-hosted-ui-in-android-app) to apply this fix :)
